static/js/hashchange.js

```diff
@@ -311,6 +311,14 @@ exports.test_regex_in_current_hash = function (regex) {
     return regex.test(window.location.hash);
 }
 
+exports.get_hash_safe = function () {
+    if (typeof window !== "undefined" && typeof window.location.hash === "string") {
+        return window.location.hash.slice(1);
+    }
+
+    return "";
+}
```

Comment:
I think this can be deleted and we can use `hash_util.get_hash_section`, or something similar instead.

Maybe most of these are just places we should have converted to use `hash_util.js` and forgot to.